holmes/plugins/toolsets/coralogix/utils.py

```diff
@@ -0,0 +1,166 @@
+   code: 'MODULE_NOT_FOUND',
+2025-03-25T07:43:39.062945626Z etc.
+```
+
+this method will print:
+
+```
+2025-03-25T07:43:39.062945526Z Require stack:
+                                   at defaultResolveImpl (node:internal/modules/cjs/loader:1061:19)
+                                   at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1066:22)
+                                   at Module.require (node:internal/modules/cjs/loader:1491:12)
+                                 code: 'MODULE_NOT_FOUND',
+2025-03-25T07:43:39.062945626Z etc.
+```
+
+
+"""
+
+
+def indent_multiline_log_message(indent_char_count: int, log_message: str):
+    lines = log_message.replace("\r", "\n").split("\n")
+    log_message = lines.pop(0)
+    while (
+        not log_message and len(lines) > 0
+    ):  # Some log messages start with a line feed or return carriage. Make sure the first line is not empty
+        log_message = lines.pop(0)
+    for new_line in lines:
+        if new_line.strip():
+            line = "\n" + (" " * indent_char_count) + new_line
+            log_message += line
+    return log_message
+
+
+def normalize_datetime(date_str: str) -> str:
+    if not date_str:
+        return "UNKNOWN_TIMESTAMP"
+
+    try:
+        date = datetime.fromisoformat(date_str.rstrip("Z"))
+        # Ensure the timestamp has the maximum resolution (nanoseconds)
+        normalized_date_time = date.strftime("%Y-%m-%dT%H:%M:%S.%f")
+        return normalized_date_time + "Z"
+    except Exception:
+        logging.debug(f"Failed to normalize timestamp {date_str}")
+        return date_str
+
+
+def extract_logs(
+    json_objects: List[Dict[str, Any]], add_namespace_tag: bool, add_pod_tag: bool
+) -> str:
+    """Extracts timestamp and log values from parsed JSON objects, sorted in ascending order (oldest first)."""
+    logs = []
+
+    for data in json_objects:
+        if isinstance(data, dict) and "result" in data and "results" in data["result"]:
+            for entry in data["result"]["results"]:
+                try:
+                    user_data = json.loads(entry.get("userData", "{}"))
+                    kubernetes = user_data.get("kubernetes", None)
+                    timestamp = normalize_datetime(user_data.get("time"))
+                    log_message = user_data.get("log", "")
+                    tags = format_kubernetes_info(
+                        kubernetes, add_namespace_tag, add_pod_tag
+                    )
+                    if log_message:
+                        logs.append(
+                            (timestamp, log_message, tags)
+                        )  # Store as tuple for sorting
+
+                except json.JSONDecodeError:
+                    logging.error(
+                        f"Failed to decode userData JSON: {entry.get('userData')}"
+                    )
+        else:
+            logging.error(f"{data}")
+
+    logs.sort(key=lambda x: x[0])
+
+    formatted_logs = []
```

Comment:
Consider to extract to method.

Reply:
yes, done as part of addressing the previous comment